src/api/v1/controllers/authController.js

```diff
@@ -3,5 +3,28 @@
  * @param {*} req - request
  * @param {*} res - response
  */
-export const userSignup = (req, res) => {
-}
\ No newline at end of file
+
+import { checkEmailUnique } from "../middlewares/validation/userSignup.js";
+import { createUser } from "../../../models/UserModel.js";
+
+export const userSignup = async (req, res) => {
+	const { firstname, lastname, email, password } = req.body;
+
+	try {
+		const isEmailUnique = await checkEmailUnique(email);
```

Comment:
You shouldn't be doing any logic in your controllers. Your controllers are meant to be very slim with only one function - handing off operations to the Service or Model and returning a success or error response with the data or error object.